edsnlp/pipelines/core/contextual_matcher/contextual_matcher.py

```diff
@@ -0,0 +1,446 @@
+                    "attr": p.regex_attr,
+                    "flags": p.regex_flags,
+                }
+                for source, p in self.patterns.items()
+            }
+        )
+
+        self.sides = ["before", "after"]
+
+        self.exclude_matchers = {side: dict() for side in self.sides}
+        self.assign_matchers = {side: dict() for side in self.sides}
+
+        for side in self.sides:
+            for source, p in self.patterns.items():
+
+                p = p.dict()
+
+                exclude_matcher = None
+
+                if p["exclude"][side]["regex"]:
+                    exclude_matcher = RegexMatcher(
+                        attr=p["regex_attr"] or self.attr,
+                        flags=p["regex_flags"] or self.regex_flags,
+                        ignore_excluded=ignore_excluded,
+                        alignment_mode="expand",
+                    )
+                    exclude_matcher.build_patterns(
+                        regex=dict(
+                            exclude=p["exclude"][side]["regex"],
+                        )
+                    )
+
+                self.exclude_matchers[side][source] = dict(
+                    matcher=exclude_matcher,
+                    window=p["exclude"][side]["window"],
+                )
+
+                assign_matcher = None
+
+                if p["assign"][side]["regex"]:
+
+                    assign_matcher = GroupRegexMatcher(
+                        attr=p["regex_attr"] or self.attr,
+                        flags=p["regex_flags"] or self.regex_flags,
+                        ignore_excluded=ignore_excluded,
+                        alignment_mode=alignment_mode,
+                    )
+
+                    assign_matcher.build_patterns(
+                        regex=p["assign"][side]["regex"],
+                    )
+
+                self.assign_matchers[side][source] = dict(
+                    matcher=assign_matcher,
+                    window=p["assign"][side]["window"],
+                    expand_entity=p["assign"][side]["expand_entity"],
+                )
+
+        self.set_extensions()
+
+    @staticmethod
+    def set_extensions() -> None:
+        if not Span.has_extension("assigned"):
+            Span.set_extension("assigned", default=dict())
+        if not Span.has_extension("source"):
+            Span.set_extension("source", default=None)
+
+    def process(self, doc: Doc) -> List[Span]:
+        """
+        Process the document, looking for named entities.
+
+        Parameters
+        ----------
+        doc : Doc
+            spaCy Doc object
+
+        Returns
+        -------
+        List[Span]
+            List of detected spans.
+        """
+
+        matches = self.phrase_matcher(doc, as_spans=True)
+        regex_matches = self.regex_matcher(doc, as_spans=True)
+
+        spans = list(matches) + list(regex_matches)
+        spans = list(self.filter(spans))
+        spans = list(self.assign(spans))
```

Comment:
Do we need these lists ?
Couldn't we instead do:
```suggestion
        spans = (*matches, *regex_matches)
        spans = self.filter(spans)
        spans = self.assign(spans)
```